code/modules/projectiles/gun.dm

```diff
@@ -166,12 +169,14 @@
 			step(user, user.inertia_dir)
 
 	if(silenced)
-		playsound(user, fire_sound, 10, 1)
+		if(fire_sound)
+			playsound(user, fire_sound, 10, 1)
 	else
-		playsound(user, fire_sound, 50, 1)
-		user.visible_message("<span class='warning'>[user] fires [src][reflex ? " by reflex":""]!</span>", \
-		"<span class='warning'>You fire [src][reflex ? "by reflex":""]!</span>", \
-		"You hear a [istype(in_chamber, /obj/item/projectile/beam) ? "laser blast" : "gunshot"]!")
+		if(fire_sound)
+			playsound(user, fire_sound, 50, 1)
+			user.visible_message("<span class='warning'>[user] fires [src][reflex ? " by reflex":""]!</span>", \
```

Comment:
Except that is what the silenced var is for, as clearly seen just a few lines up